### PR TITLE
Adding SMS usage report as a S3 data source to QuickSight

### PR DESCRIPTION
**Pull request by jimleroyer**
# Summary | Résumé

With the goal of tracking the cost per service, these changes add the S3 usage report into QuickSight as a data source. 

Remaining work after this PR:

* Adding the SMS usage report as a data set. We might try through clickops first.
* Joining the data set of the SMS usage report with the notifications data set, using the `reference` and `Message ID` fields for each.
* Restrict IAM access as privileges are wide at the moment.

# Test instructions | Instructions pour tester la modification

Most likely these changes will require further ones, as access might not be valid and we will need to analyze these. Verify properly the Terraform plan and make sure the apply does not fail. If it does, revert or fix ASAP.

### CODE REVIEW

File: aws/quicksight/dataset_notifications.tf [hunk 1]
```diff
@@ -102,6 +105,10 @@ resource "aws_quicksight_data_set" "notifications" {
         name = "notification_type"
         type = "STRING"
       }
+      columns {
+        name = "notification_resource"
```
Comment (sastels) [inline aws/quicksight/dataset_notifications.tf:109]:
```suggestion
        name = "notification_reference"
```

**Review by sastels — CHANGES_REQUESTED** [review]
Just that one typo then LGTM!

File: aws/quicksight/iam.tf [hunk 2]
```diff
@@ -42,3 +42,28 @@
...
+          "s3:GetObjectVersion",
+          "s3:ListBucket",
+          "s3:ListBucketMultipartUploads"
+        ]
+        Effect   = "Allow"
+        Resource = "*"
+      },
+    ]
+  })
+}
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/quicksight/iam.tf:64]: we are planning on restricting these once we get it working right?

[the remark this answers is not in the document]
Comment (jimleroyer) [reply aws/quicksight/iam.tf:64]: Yes! in the following days.

File: aws/quicksight/dataset_notifications.tf (hunk 1 above)
Comment (jimleroyer) [reply aws/quicksight/dataset_notifications.tf:109]: I probably made that typo. 🤚

**Comment by jimleroyer** [comment]
Merging, and refining security complaint as next step!

**Pull request merged**